File: July 2025 GFG SOLUTION/July-22.java

class Solution {
    public int missingNumber(int[] a) {
        int n = a.length;
        for (int i = 0; i < n; i++) if (a[i] <= 0) a[i] = n + 1;
        for (int i = 0; i < n; i++) {
            int x = Math.abs(a[i]);
            if (x <= n) a[x - 1] = -Math.abs(a[x - 1]);
        }
        for (int i = 0; i < n; i++) if (a[i] > 0) return i + 1;
        return n + 1;
    }
}

File: July 2025 GFG SOLUTION/July-22.py
class Solution:
    def missingNumber(self, a):
        n = len(a)
        for i in range(n):
            while 0 < a[i] <= n and a[i] != a[a[i] - 1]:
                a[a[i] - 1], a[i] = a[i], a[a[i] - 1]
        for i in range(n):
            if a[i] != i + 1: return i + 1
        return n + 1

2)
class Solution:
    def missingNumber(self, a):
        n = len(a)
        for i in range(n):
            if a[i] <= 0: a[i] = n + 1
        for i in range(n):
            x = abs(a[i])
            if x <= n: a[x - 1] = -abs(a[x - 1])
        for i in range(n):
            if a[i] > 0: return i + 1
        return n + 1
